tests/unit/FirebaseStorage.spec.js

```diff
@@ -0,0 +1,361 @@
+'use strict';
+describe('$firebaseStorage', function () {
+  var $firebaseStorage;
+  var URL = 'https://oss-test.firebaseio.com';
+
+  beforeEach(function () {
+    module('firebase.storage');
+  });
+
+  describe('<constructor>', function () {
+
+    var $firebaseStorage;
+    var $q;
+    var $rootScope;
+    var $firebaseUtils;
+    beforeEach(function () {
+      module('firebase.storage');
+      inject(function (_$firebaseStorage_, _$q_, _$rootScope_, _$firebaseUtils_) {
+        $firebaseStorage = _$firebaseStorage_;
+        $q = _$q_;
+        $rootScope = _$rootScope_;
+        $firebaseUtils = _$firebaseUtils_;
+      });
+    });
+
+    it('should exist', inject(function () {
+      expect($firebaseStorage).not.toBe(null);
+    }));
+
+    it('should create an instance', function () {
+      var ref = firebase.storage().ref('thing');
+      var storage = $firebaseStorage(ref);
+      expect(storage).not.toBe(null);
+    });
+
+    describe('$firebaseStorage.utils', function () {
+
+      describe('_unwrapStorageSnapshot', function () {
+
+        it('should unwrap the snapshot', function () {
+          var mockSnapshot = {
+            bytesTransferred: 0,
+            downloadURL: 'url',
+            metadata: {},
+            ref: {},
+            state: {},
+            task: {},
+            totalBytes: 0
+          };
+          var unwrapped = $firebaseStorage.utils._unwrapStorageSnapshot(mockSnapshot);
+          expect(mockSnapshot).toEqual(unwrapped);
+        });
+
+      });
+
+      describe('_$put', function () {
+
+        it('should call a storage ref put', function () {
+          var ref = firebase.storage().ref('thing');
+          var file = 'file';
+          var task = null;
+          var digestFn = $firebaseUtils.compile;
+          spyOn(ref, 'put');
+          task = $firebaseStorage.utils._$put(ref, file, digestFn, $q);
+          expect(ref.put).toHaveBeenCalledWith('file');
+          expect(task.$progress).toEqual(jasmine.any(Function));
```

Comment:
I think these last three lines are not needed in this test since they are covered in the next test.